### PR TITLE
Added backgroundColor property to NSView extension

### PR DESCRIPTION
**Pull request by phiren**
Added backgroundColor property to NSView extension
No new file, added to existing file
It will help to set background color of NSView and it will also help to get background color of NSView

### CODE REVIEW

File: Sources/Extensions/AppKit/NSViewExtensions.swift [hunk 1]
```diff
@@ -141,4 +141,23 @@ extension NSView {
 	}
 
 }
+
+extension NSView
+{
+    /// SwifterSwift: set and get background color of NSView
+    @IBInspectable public var backgroundColor: NSColor? {
```
Comment (SD10) [inline Sources/Extensions/AppKit/NSViewExtensions.swift]: @phiren Can you move this to be up above with the other properties?

**Comment by SD10** [comment]
This PR is also missing:
- [ ] Unit tests
- [ ] CHANGELOG entry

Please see the [SwifterSwift contributing guidelines](https://github.com/SwifterSwift/SwifterSwift/blob/master/CONTRIBUTING.md)

**Comment by phiren** [comment]
Hi.

Can you help me for adding unit tests and changeling entry?
Never done before

Thanks,
Hiren

File: Sources/Extensions/AppKit/NSViewExtensions.swift (hunk 1 above)
Comment (phiren) [reply Sources/Extensions/AppKit/NSViewExtensions.swift]: @SD10 , sure, but can you help me for adding unit test cases and change log?

**Comment by BennX** [comment]
@phiren thank you for adding the changelog and the unittests. The tests didn't pass for MacOS. Could you please fix the tests for the MacOS?  https://travis-ci.org/SwifterSwift/SwifterSwift/jobs/412149217#L3045

File: Sources/Extensions/AppKit/NSViewExtensions.swift [hunk 2]
```diff
@@ -101,6 +101,21 @@ public extension NSView {
 			layer?.shadowRadius = newValue
 		}
 	}
+    
+    /// SwifterSwift: set and get background color of NSView
+    @IBInspectable public var backgroundColor: NSColor? {
+        get {
+            if let colorRef = self.layer?.backgroundColor {
```
Comment (VincentSit) [inline Sources/Extensions/AppKit/NSViewExtensions.swift]: I'm not reviewer, but I have a suggestion. You should use `self` only when required by the compiler. This library also follows this style.

Comment (phiren) [reply Sources/Extensions/AppKit/NSViewExtensions.swift]: hi @VincentSit , removed self, thanks for suggestion!
I never used this standard's before, Can someone help here for understanding how all it works!
I see how codecove and swifter bot for different tasks, I never used it.

File: Sources/Extensions/AppKit/NSViewExtensions.swift [hunk 3]
```diff
@@ -101,6 +101,21 @@ public extension NSView {
 			layer?.shadowRadius = newValue
 		}
 	}
+    
+    /// SwifterSwift: set and get background color of NSView
```
Comment (LucianoPAlmeida) [inline Sources/Extensions/AppKit/NSViewExtensions.swift]: Minor here, but can you change "set" to "Set" on this docs :))

**Comment by LucianoPAlmeida** [comment]
@phiren Can you fix the swiftlint warnigs pointed by danger? :))
I think a swiftlint autocorrect should do it 👍

**Comment by phiren** [comment]
@LucianoPAlmeida , Not able to install swiftlint as extension support removed from Xcode 8, let me know if there is any way to install extension

**Comment by guykogus** [comment]
You install it with [Homebrew](https://brew.sh)

File: Tests/AppKitTests/NSViewExtensionsTests.swift [hunk 4]
```diff
@@ -123,6 +123,18 @@ final class NSViewExtensionsTests: XCTestCase {
 		view.removeSubviews()
 		XCTAssertEqual(view.subviews.count, 0)
 	}
+
```
Comment (guykogus) [inline Tests/AppKitTests/NSViewExtensionsTests.swift]: These are trailing white spaces. You should get SwiftLint warnings about these. You can change your Xcode settings to clear them for you. In your Xcode properties go to the **Text Editing** section and enable **Automatically trim trailing whitespace**, along with **Including whitespace-only lines** as well.

File: Sources/Extensions/AppKit/NSViewExtensions.swift [hunk 5]
```diff
@@ -12,133 +12,149 @@ import Cocoa
...
+        get {
+            return layer?.shadowRadius ?? 0
+        }
+        set {
+            wantsLayer = true
+            layer?.shadowRadius = newValue
+        }
+    }
+
+    /// SwifterSwift: Set and get background color of NSView
```
Comment (guykogus) [inline Sources/Extensions/AppKit/NSViewExtensions.swift]: This comment should be
```Swift
/// SwifterSwift: Background color of the view; also inspectable from Storyboard.
```
That way it's more consistent with the other property comments.

**Comment by guykogus** [comment]
@phiren there are a couple of small comments here. Can you please fix them and then we can merge in your change? :)

**Comment by LucianoPAlmeida** [comment]
Closing this due inactivity
Feel free to reopen it anytime :))

**Comment by RomanPodymov** [comment]
Fixes #533.

**Pull request closed without merging**